tutorials/time-freq/plot_ssvep.py

```diff
@@ -0,0 +1,767 @@
+=======================================
+Basic analysis of an SSVEP/vSSR dataset
+=======================================
+
+In this tutorial we compute the frequency spectrum and quantify signal-to-noise
+ratio (SNR) at a target frequency in EEG data recorded during fast periodic
+visual stimulation (FPVS).
+Extracting SNR at stimulation frequency is a simple way to quantify frequency
+tagged responses in MEEG (a.k.a. steady state visually evoked potentials,
+SSVEP, or visual steady-state responses, vSSR in the visual domain,
+or auditory steady-state responses, ASSR in the auditory domain).
+
+DATA:
+
+We use a simple example dataset with frequency tagged visual stimulation:
+N=2 participants observed checkerboards patterns inverting with a constant
+frequency of either 12Hz of 15Hz. 10 trials of 20s length each.
+32ch wet EEG was recorded.
+
+Data can be downloaded at https://osf.io/7ne6y/
+Data format: BrainVision .eeg/.vhdr/.vmrk files organized according to BIDS
+standard.
+
+OUTLINE:
+
+- We will visualize both the PSD and the SNR spectrum of our epoched data.
+
+- We will extract SNR at stimulation frequency for all trials and channels.
+
+- We will show, that we can statistically separate 12hz and 15 hz responses
+  in our data.
+
+
+Since the evoked response is mainly generated in early visual areas of the
+brain we will stick with an ROI analysis and extract SNR from occipital
+channels.
+"""  # noqa: E501
+# Authors: Dominik Welke <dominik.welke@web.de>
+#          Evgenii Kalenkovich <e.kalenkovich@gmail.com>
+#
+# License: BSD (3-clause)
+
+import matplotlib.pyplot as plt
+import mne
+import numpy as np
+from scipy.stats import ttest_rel
+
+###############################################################################
+# Data preprocessing
+# ------------------
+# Due to a generally high SNR in SSVEP/vSSR, typical preprocessing steps
+# are considered optional. This doesn't mean, that a proper cleaning would not
+# increase your signal quality!
+#
+# * Raw data have FCz reference, so we will apply common-average rereferencing.
+#
+# * We will apply a 50 Hz notch filter to remove line noise,
+#
+# * and a 0.1 - 250 Hz bandpass filter.
+#
+# * Lastly, we will cut the data in 20 s epochs corresponding to the trials.
+#
+#
+
+# Load raw data
+data_path = mne.datasets.ssvep.data_path()
+bids_fname = data_path + '/sub-02/ses-01/eeg/sub-02_ses-01_task-ssvep_eeg.vhdr'
+
+raw = mne.io.read_raw_brainvision(bids_fname, preload=True, verbose=False)
+raw.info['line_freq'] = 50.
+
+# Set montage
+montage_style = 'easycap-M1'
+montage = mne.channels.make_standard_montage(
+    montage_style,
+    head_size=0.095)  # head_size parameter default = 0.095
+raw.set_montage(montage, verbose=False)
+
+# Set common average reference
+raw.set_eeg_reference('average', projection=False, verbose=False)
+
+# Apply notch filter to remove line-noise
+notch = np.arange(raw.info['line_freq'], raw.info['lowpass'] / 2,
+                  raw.info['line_freq'])
```

Comment:
```suggestion
notch = np.arange(raw.info['line_freq'], raw.info['sfreq'] / 2,
                  raw.info['line_freq'])
```

Reply:
included in https://github.com/mne-tools/mne-python/pull/8867/commits/e0bbb2b609efc7f28224b86f3f6d125578184142